backend/server.js

```diff
@@ -116,25 +117,37 @@ const sendGFIEmail = async (repoURL) => {
       })
       console.log("Couldn't find subscribed repo: ", error);
     }
-  } catch(error) {
+  } catch (error) {
     console.error('Error fetching GFIs:', error);
   }
 };
 
 const repoToCheck = [];
 const addRepoToCheck = (repoURL) => {
-  if(!repoToCheck.includes(repoURL)) {
+  if (!repoToCheck.includes(repoURL)) {
     repoToCheck.push(repoURL);
     console.log(`Added ${repoURL} to repoToCheck.`);
   }
 };
 
-const cronJob = new cron.CronJob('* * * * *', () => {
+const cronJob = new cron.CronJob('* * * * *', async () => {
+  
   console.log('Checking for new GFIs...');
-  repoToCheck.forEach((repoURL) => {
-    sendGFIEmail(repoURL);
-    console.log(`Trying to send an email for ${repoURL}.`)
-  });
+  try {
+    if (repoToCheck.length == 0) {
+      let repoData = await Subscription.find({});
+      repoData = [...new Set(repoData.map(item => item.repourl))];
```

Comment:
here change the `repourl` ro `repoURL` as you're fetching the item object with the `repoURL` field.